### DISCUSSION TITLE
Edge Functions / VueJS / App Hosting

### DISCUSSION THREAD

**Discussion by SteveChurch** [supabase/supabase#11619] [Questions] [answered]
I'm working on a VueJS project and have been using Supabase. Firstly, Kudos, it's great. 

However, I feel there is one thing missing... App hosting. 

You have Edge Functions which are great, but obviously, they are Deno and primarily for.. well Edge functions, not running a whole web framework. 

Is this something that is in the pipeline I love everything in one place, or am I missing a feature I can't find :D 

Thanks

**Comment by GaryAustin1** [comment, marked as answer]
This is a 8 month old blog post but gives the feel at the time about hosting... https://supabase.com/blog/supabase-edge-functions

> Can I host static sites

> Even though Deno supports [rendering jsx and tsx files](https://deno.land/manual/jsx_dom/jsx), we don't plan on becoming a website hosting provider with Edge Functions. For now, Edge Functions can only be invoked via the POST method. **Offering secure and fast web hosting with a good developer experience is a hard problem to solve and there are [other](https://www.netlify.com/) [fantastic](https://vercel.com/) [providers](https://pages.cloudflare.com/) which do exactly that.**
> 
> **From our experience launching Supabase Storage last year, becoming a hosting provider is not something you want to take lightly.** Malicious folks will upload their phishing sites whenever they get a chance and this can quickly spiral out of control, causing you to land on the denylist in the [Safe Browsing dataset](https://safebrowsing.google.com/). Even though we removed the project within minutes after being notified, various ISPs had already cached the information from the Safe browsing dataset and were “helping” their users by restricting access to the Supabase domain. This isn't something we're planning to repeat. But as always, we'd love to hear your use cases and feature requests, we're just one [GitHub discussion](https://github.com/supabase/supabase/discussions) away!

So far past experience shows they do not normally announce future plans before Launch Weeks.

**Reply by vipulb2** [reply]
This seems important. Beyond just the usual problems described in this blog post, there are several other issues. Keeping hackers away is the least of these problems.!:)

Providing staticc WEb-hosting is something these folks at Supabase will work on, we hope.🙏

**Comment by tohagan** [comment]
Probably the most compelling reason I've stayed with Firebase and avoided Superbase is the lack of ability to perform deployment of the full stack including static hosting of the Vue/React front end on CDN. Given that your already have a nice CDN it seems so crazy that you still don't have a full stack deployment. When you add this missing piece I'll join you.  

Here's what this and many other customers want (must haves). Trigger or command to deploy: 
- Build and push static hosted SPA/PWA UI to CD, linked to custom domain has LetsEncrypt SSL on the custom domain
  - No ... I don't need/want edge functions to host the static content
- Rest API or GraphQL API linked to same custom domain to simplify CORS
- Database migrations auto applied
Nice to have:
- GIT versioned so can be linked to a Preview review => Staging release => Prod release

Questions:
- How far off this mark are we now?
- Will it ever happen?
- Is your commication missing thet fact that you already offer this?
